frontend/taipy-gui/src/components/Taipy/ProgressBar.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2021-2024 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+import React from "react";
+import {
+    Box,
+    CircularProgress,
+    CircularProgressProps,
+    LinearProgress,
+    LinearProgressProps,
+    Typography,
+} from "@mui/material";
+import { useEffect, useState } from "react";
+import { useDynamicProperty } from "../../utils/hooks";
+
+interface ProgressBarProps {
+    linear?: boolean; //by default - false
+    showProgress?: boolean; //by default - false
+    value?: number; //progress value
+    defaultValue?: number; //default progress value
+}
+
+const Progress = (props: ProgressBarProps) => {
+    const { linear, showProgress, value, defaultValue } = props;
+
+    const [linearProgress, setLinearProgress] = useState(false);
```

Comment:
no need to useState for those 2 boolean values

Reply:
But, without `useState`, we'll not be able to check whether the dev wants the progress to be linear or circular one?

Reply:
No, actually I was talking about the updating part. The `linear` prop is be default set to `false`. But, if the dev wants the linear element, then the dev must enter a param like this - `return progress(linear=True)` and the linear param will be updated only if we use the `useState` and the `useEffect` hook. I'm not able to understand that how can it be done without these 2 hooks?

Reply:
taipy will take care of that: updating the prop

Reply:
So, should I remove both the hooks and then simply use the `props.linear` in the `if/else` block, right?